relay-metrics/src/aggregation.rs

```diff
@@ -1176,24 +1178,18 @@ impl Message for FlushBuckets {
 }
 
 /// Check whether the aggregator has not (yet) exceeded its total limits. Used for health checks.
+#[derive(Debug)]
 pub struct AcceptsMetrics;
 
 impl Message for AcceptsMetrics {
     type Result = bool;
 }
 
-impl Handler<AcceptsMetrics> for Aggregator {
-    type Result = bool;
-
-    fn handle(&mut self, _msg: AcceptsMetrics, _ctx: &mut Self::Context) -> Self::Result {
-        !self
-            .cost_tracker
-            .totals_cost_exceeded(self.config.max_total_bucket_bytes)
-    }
-}
-
 enum AggregatorState {
     Running,
+    // NOTE: This isn't used at the moment, since this service does not handle the shutdown
+    // messages just yet.
+    #[allow(dead_code)]
```

Comment:
You need to use `Controller::subscribe_v2` and use the watch channel: see https://github.com/getsentry/relay/blob/95f154409f236ecc8089a4b19534997625a21735/relay-server/src/actors/health_check.rs#L129-L135 for an example of how we're currently thinking of handling this.  (though honestly that whole system probably needs changing as part of the more structured shutdown, but for now this is probably the way to do this)

Reply:
see the comment in https://github.com/getsentry/relay/pull/1508#discussion_r985744067